src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJsonAndInstalledJson.php

```diff
@@ -38,19 +38,22 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
             throw InvalidProjectDirectory::atPath($installationPath);
         }
 
-        $composerJsonPath  = $realInstallationPath . '/composer.json';
-        $installedJsonPath = $realInstallationPath . '/vendor/composer/installed.json';
+        $composerJsonPath = $realInstallationPath . '/composer.json';
 
         if (! file_exists($composerJsonPath)) {
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
+        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}, config: array{vendor-dir: array<string, array<int, string>>}}|null $composer */
```

Comment:
Can we split this into multiline?

```php
/**
 * @var array{
 * ...
 * }
 */
```

Reply:
`vendor-dir` is incorrectly mapped here.

Currently: `vendor-dir: array<string, array<int, string>>`

Correct: `vendor-dir?: string`

Reply:
The `phpcs` complaines about the multiline annotation. 

``` 
 49 | ERROR | Invalid inline documentation comment format "@var array{
    |       |          * autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int,
    |       | string>>, psr-0: array<string, array<int, string>>},
    |       |          * config: array{vendor-dir?: string}
    |       |          * }|null $composer", expected "@var type $variable".
    |       | (SlevomatCodingStandard.Commenting.InlineDocCommentDeclaration.InvalidFormat)

``` 
But don't know how to deal with it.

Reply:
Oh, use `/** @psalm-var` and phpcs will shut up :-)

Reply:
If you say so, didn't want to ignore it till getting feedback. I'm trying to learn and contribute, not to screw your library :)